### ISSUE TITLE
Allow users to sign-up / invite others by phone number

### ISSUE THREAD

**Issue by nbiton** [unee-t/frontend#527]
I guess many users who are used to communicating about their property over **Whatsapp** will find it hard to switch over to Unee-T, as they don't have the email addresses of all the people they are used to communicate with.
Allowing using the phone number, at least for invitations, will remove a lot of friction from the process onboarding new users.

**Comment by franck-boullier** [comment]
@nbiton that would be a great feature yes.

There are some minor issues with that
- We'll have to create a 'dummy email' for the user with only a phone number
- We'll have to create a different 'reset password' flow for the user with only phone numbers (we could send these user a SMS instead)

Apart from that I agree we'll need that feature in the future

**Comment by franck-boullier** [comment]
@kaihendry how complex will it be to have a solution to send SMS notifications for these invitations?
@kiatlim can you help us designing the flow for that scenario (we invite a user via SMS instead of email)?

**Comment by kaihendry** [comment]
It's straightforward to send an SMS from AWS. Though there are so many little limitations wrt encoding and limits that, ultimately I feel it's a total PITA / waste of time.

    aws --profile uneet-dev sns publish --phone-number +65xxxxxxx --message "hello from Unee-T dev"

Handling the edge cases of a user account with just a phone number will be quite tricky and not worth dealing with. I would prioritise account Google logins instead. You need a Google login to use an Android phone for example.

**Comment by franck-boullier** [comment]
> It's straightforward to send an SMS from AWS

OK thanks.

> Though there are so many little limitations wrt encoding and limits

That should be manageable since we will totally control what we send. 
We can decide to only send minified URLs via SMS for example. User receives the SMS and clicks on the link et voila...

> Handling the edge cases of a user account with just a phone number will be quite tricky

Not necessarily...
An idea would be to create a email for the users who just have a phone number in their profile.
That email will be used to forward critical messages (like password reset for instance) to the phone number for that user via SMS.
We'll only have to maintain a mapping between the auto generated email @case.unee-t.com and the phone number for that user.

**Comment by ongdominic** [comment]
We can add this additional field to the invite page. After that, the user will receive an sms with the minified URL so create an account or log in. Does this mean we will need to store and show the user's mobile number on their profile page also? 

![screenshot 2019-01-22 at 2 39 03 pm](https://user-images.githubusercontent.com/46714451/51516789-7ffe4580-1e53-11e9-9474-872f34e17cc6.png)

**Comment by ongdominic** [comment]
Yup. you are right. I have done the adjustments. I don't think there will be a use case whereby you want to invite by both email and phone. 

![screenshot 2019-01-22 at 4 02 36 pm](https://user-images.githubusercontent.com/46714451/51520661-43d0e200-1e5f-11e9-9682-504778a10220.png)

**Comment by kaihendry** [comment]
Made an [email2sms demo](https://media.dev.unee-t.com/2019-01-25/email2sms.mp4) to demonstrate how we can implement the sms sending via an existing email implementation. It's disabled right now to prevent abuse.
